src/main/java/com/github/rodionovsasha/commentservice/services/TopicService.java

```diff
@@ -0,0 +1,11 @@
+package com.github.rodionovsasha.commentservice.services;
+
+import com.github.rodionovsasha.commentservice.entities.Topic;
+import com.github.rodionovsasha.commentservice.exceptions.TopicNotFoundException;
+
+public interface TopicService {
```

Comment:
Let's make `TopicService` more real and more UI oriented:

```java
// Creates a new topic unless the user is inactive
// @throws UserNotFoundException when user was not found by id
// @throws InactiveUserException when user is found but he or she is not active
Topic startTopic(long userId, String title) throws ...

// Deletes a topic. The deletion of already deleted topic does not do anything.
// @throws TopicNotFoundException when topic is not found
// @throws UserNotFoundException
// @throws InactiveUserException
// @throws PermissionException when a non topic starter is trying to delete the topic
void deleteTopic(long topicId, long userId) throws ...;

// Lists topics which belong to a user
// Each topic entity should not get comments from DB nor serialize them to JSON
// Getting topics of inactive users is allowed
// Only not deleted topic should be returned
// @throws UserNotFoundException
List<Topic> listForUser(long userId, org.springframework.data.domain.Sort sort) throws ...;

// Searched for not deleted topics if title includes a fragment(case insensitive)
// Results should be limited by `limit` 
List<Topic> search(String titleFragment, int limit);
```

I suggest to comment out the TopicController, then to implement this real service and then to test it properly